main.py

```diff
@@ -59,186 +60,193 @@ def main():
-
-        if announcement_coin not in order and announcement_coin not in sold_coins and announcement_coin not in old_coins:
-            if announcement_coin:
+            if announcement_coin and announcement_coin not in order and announcement_coin not in sold_coins and announcement_coin not in old_coins:
                 logger.info(f'New announcement detected: {announcement_coin}')
-            if supported_currencies:
-                if announcement_coin and announcement_coin in supported_currencies:
-                    logger.debug("Starting get_last_price")
-                    price = get_last_price(announcement_coin, pairing)
-
-                    logger.debug('Coin price: ' + price)
-                    logger.debug('Finished get_last_price')
-
-                    try:
-                        # Run a test trade if true
-                        if config['TRADE_OPTIONS']['TEST']:
-                            order[announcement_coin] = {
-                                'symbol': announcement_coin,
-                                'price': price,
-                                'volume': qty,
-                                'time': datetime.timestamp(datetime.now()),
-                                'tp': tp,
-                                'sl': sl,
-                                'id': 'test-order',
-                                'text': 'test-order',
-                                'create_time': datetime.timestamp(datetime.now()),
-                                'update_time': datetime.timestamp(datetime.now()),
-                                'currency_pair': f'{announcement_coin}_{pairing}',
-                                'status': 'filled',
-                                'type': 'limit',
-                                'account': 'spot',
-                                'side': 'buy',
-                                'iceberg': '0'
-                            }
-                            logger.info('PLACING TEST ORDER')
-                            logger.debug(order[announcement_coin])
-                        # place a live order if False
-                        else:
-                            logger.info(f'starting buy place_order with : {announcement_coin=} | {pairing=} | {qty=} | side = buy | {price=}')
-                            order[announcement_coin] = place_order(announcement_coin, pairing, qty,'buy', price)
-                            order[announcement_coin] = order[announcement_coin].__dict__
-                            order[announcement_coin].pop("local_vars_configuration")
-                            order[announcement_coin]['tp'] = tp
-                            order[announcement_coin]['sl'] = sl
-                            logger.info('Finished buy place_order')
-
-                    except Exception as e:
-                        logger.error(e)
+                if not supported_currencies:
+                    supported_currencies = get_all_currencies(single=True)
```

Comment:
If supported_currencies is False, we still want to try to make the buy order this loop, so we should try to update it and then see if it's set again.